src/App.js

```diff
@@ -1,16 +1,26 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [likeCount, setLikeCount] = useState(0);
+  const updateLikeCount = (increment) => {
+    setLikeCount(likeCount + increment);
+  };
```

Comment:
I was about to comment that there was no need to have a `likeCount` state variable, but then I realized that this rather cleverly avoids having a state variable of all the chat entries. There is a slight downside, though...

Right now all the entries in messages.json have `liked` set to `false`, but if we were to imagine that rather taking this from static data we would get this data from an API, it would be possible for some of the entries to have already been liked and thus have a value of `true` for `liked`.

However, with this code, those likes would not appear in this number, and in fact, if you unliked the message, the number would go negative.

Of course, there's a simple solution to that, and that's to count the likes before making your call to `useState`, but at the same time, if we _were_ taking in a list of messages from an API, we probably _would_ have to actually maintain the chat entries using `useState`.

And in that case, rather than having two state variables, one for the entries and one for the count of likes, it would likely be better to only maintain the one for the entries as the number of likes can be determined from the entries.

Having redundant state like that is a common source for bugs as the state that _should_ stay in sync falls out of sync. Not necessarily enough to be a bug, but often what a reviewer would call a "code smell".

But we're well in hypothetical world at this point, just wanted to point that out. I think for our case your implementation is fine!

Reply:
Interested in seeing what happened, I cloned your code and set one the messages to have a `liked` of `true` and to my surprise (because I haven't looked at `ChatEntry` yet, that message was displayed as not liked. Which, again, would be fine without my changes, but everything I said in this comment would apply to the heart widget in `ChatEntry` too.